### PR TITLE
feature: make it works without submodules

### PR DESCRIPTION
**Pull request by ClausKlein**
use CMP.cmake
cleanup cmake files
prevent warnings, but not all
setup GitHub Actions for CI see https://github.com/ClausKlein/cxx.simplelog/runs/1998057440?check_suite_focus=true

this fix 
https://github.com/jenisys/cxx.simplelog/issues/2
https://github.com/jenisys/cxx.simplelog/issues/3
https://github.com/jenisys/cxx.simplelog/issues/4

and some of (https://github.com/jenisys/cxx.simplelog/issues/6)

### CODE REVIEW

**Comment by ClausKlein** [comment]
@jenisys you should enable the GitHub Actions under Settings!

File: .cmake/project-config.cmake.in [hunk 1]
```diff
@@ -1,6 +1,9 @@
 @PACKAGE_INIT@
 
 include(CMakeFindDependencyMacro)
-find_dependency(spdlog @SPDLOG_MIN_VERSION@ REQUIRED COMPONENTS spdlog)
+#
+# TODO: find_dependency(spdlog @SPDLOG_MIN_VERSION@ REQUIRED COMPONENTS spdlog)
```
Comment (ClausKlein) [inline .cmake/project-config.cmake.in]: this should be addressed!

see https://discourse.cmake.org/t/how-to-install-component-libraries-like-boost/2854/2

File: cmake/cxx.setup_cxx_standard.cmake [hunk 2]
```diff
@@ -30,7 +30,7 @@ endif()
 # SANE DEFAULTS: If ".cmake_project.cxx_standard" is missing
 # ---------------------------------------------------------------------------
 if(NOT DEFINED CMAKE_CXX_STANDARD)
-    set(CMAKE_CXX_STANDARD 17)  # Enable C++17 standard
+    set(CMAKE_CXX_STANDARD 20) # Enable C++20 standard
```
Comment (ClausKlein) [inline cmake/cxx.setup_cxx_standard.cmake:33]: maybe we should not change this?

File: examples/CMakeLists.txt [hunk 3]
```diff
@@ -2,147 +2,125 @@
...
+
+    list(INSERT CMAKE_MODULE_PATH 0 ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
+    include(ccache)
+    include(cxx.setup_project)
+
+    # TODO: Quickfix only, should not be necessary! CK
+    find_package(simplelog)
+    # FIXME: The following imported targets are referenced, but are missing: cxx_simplelog::simplelog
+
+    find_package(simplelog_null)
```
Comment (ClausKlein) [inline examples/CMakeLists.txt]: FIXME: that should be enough!

File: src/simplelog/backend/null/CMakeLists.txt [hunk 4]
```diff
@@ -5,15 +5,24 @@
...
+packageProject(
+    NAME simplelog_null
+    VERSION ${PROJECT_VERSION}
+    NAMESPACE ${PROJECT_NAMESPACE}
+    BINARY_DIR ${PROJECT_BINARY_DIR}
+    INCLUDE_DIR ${SIMPLELOG_INCLUDE_DIR}
+    INCLUDE_DESTINATION include
+    COMPATIBILITY SameMajorVersion
+    DISABLE_VERSION_SUFFIX YES
+    # DEPENDENCIES "fmt ${FMT_MIN_VERSION}; spdlog ${SPDLOG_MIN_VERSION}"
```
Comment (ClausKlein) [inline src/simplelog/backend/null/CMakeLists.txt]: why is `simplelog_null` dependent from `simplelog`?

**Comment by ClausKlein** [comment]
@jenisys there is still an open problem: the findsyslog.cmake module is not installed!

So downstream  can't use it.

see https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right/

File: .cmake/project-config.cmake.in (hunk 1 above)
Comment (ClausKlein) [reply .cmake/project-config.cmake.in]: done

File: examples/CMakeLists.txt (hunk 3 above)
Comment (ClausKlein) [reply examples/CMakeLists.txt]: done

File: examples/CMakeLists.txt [hunk 5]
```diff
@@ -2,147 +2,119 @@
...
+
+set(MASTER_PROJECT OFF)
+if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+    set(MASTER_PROJECT ON)
+
+    list(INSERT CMAKE_MODULE_PATH 0 ${CMAKE_CURRENT_SOURCE_DIR}/../cmake)
+    include(ccache)
+    include(cxx.setup_project)
+
+    find_package(simplelog COMPONENTS null spdlog syslog)
```
Comment (ClausKlein) [inline examples/CMakeLists.txt:20]: @jenisys Note: this works now!

File: CMakeLists.txt [hunk 6]
```diff
@@ -121,51 +118,90 @@ endif()
...
 
-# -- MAYBE:
-# export(EXPORT cpp_compat NAMESPACE CppCompat)
+if(SIMPLELOG_USE_BACKEND_SYSLOG)
+    install(FILES .cmake/SyslogTargets.cmake
+            DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAMESPACE}
+    )
+endif()
+
+if(SIMPLELOG_USE_BACKEND_SYSTEMD)
```
Comment (ClausKlein) [inline CMakeLists.txt:174]: without this the install cmake config set was not useable!

**Comment by ClausKlein** [comment]
https://de.wiktionary.org/wiki/gegen_Dummheit_kämpfen_Götter_selbst_vergebens

**Comment by jenisys** [comment]
While this PR may contains some improvements and fixes it violates the best practices for an PR:

* It is not atomic: It does not address one problem or adds one feature.
  Therefore, it cannot be discussed, adapted and improved and finally accepted or rejected.
* It is not minimal: It does not contain only the changes that are really needed for this PR title/subject
* It contains too many unrelated changes (many of which are not really needed and/or not part of the subject of this PR)
* It reformats the source code and CMake files without any need
* It tries to add other stuff that should not be part of this PR
* It tries to fix multiple issues instead of one
* It tries to use clang-tidy --modernize which just causes changes w/o any real improvements
* It changes the default C++ standard which is not subject of this PR

**Pull request closed without merging**